core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,7 +64,23 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // Evaluate partitions in parallel. Partitions of each rdd will be cached by the `partitions`
+  // val in `RDD`.
+  private[spark] lazy val parallelPartitionEval: Boolean = {
```

Comment:
> I thought the issue here was use for reading/writing data rather than getSplits

The issue is introducing concurrent calls to getSplits. [PARQUET-16|https://issues.apache.org/jira/browse/PARQUET-16] is a suggestion to cache the results of file system operations during split planning, which would have to be done by a cache shared across instances. (Instead, we fixed the underlying problem with HADOOP-12810.)

> if this is a problem, this doesn't resolve it. More immediately, there are cases where the same instance of InputFormat is accessed concurrently.

If this is true, then your line of reasoning makes sense. Where are InputFormat instances accessed concurrently?